tests/scripting_api/test_window.py

```diff
@@ -0,0 +1,162 @@
+from unittest.mock import MagicMock, patch
+
+import pytest
+from hamcrest import *
+
+from autokey.scripting import Window
+
+def create_window():
+    win = Window(MagicMock())
+    return win
+
+def test_window_wait_for_focus() -> Window:
+    #TODO tests
+    pass
+
+def test_window_wait_for_exist():
+    pass
+
+def test_window_activate():
+    pass
+
+def test_window_close():
+    pass
+
+def test_window_resize_move():
+    pass
+
+def test_window_move_to_desktop():
+    pass
+
+def test_window_switch_desktop():
+    pass
+
+def test_window_set_property():
+    pass
+
+def test_window_get_active_geometry():
+    pass
+
+def test_window_get_active_title():
+    pass
+
+def test_window_get_active_class():
+    pass
+
+       #Monitors: 2
+        # 0: +*HDMI-0 1920/521x1080/293+0+0  HDMI-0
+        # 1: +DVI-D-0 1440/408x900/255+1920+0  DVI-D-0
+xrandr_raw_double = """Monitors: 2
+ 0: +*HDMI-0 1920/521x1080/293+0+0  HDMI-0
+ 1: +DVI-D-0 1440/408x900/255+1920+0  DVI-D-0"""
+xrandr_raw_single = """Monitors: 1
+ 0: +*DVI-D-0 1440/408x900/255+0+0  DVI-D-0"""
+
+@pytest.mark.parametrize("title, xrandr_raw_output, matchClass", [
+    (None, (0, xrandr_raw_single), False), #single monitor
+    (None, (0, xrandr_raw_double), False), #double monitor
+    (":ACTIVE:", (0, xrandr_raw_double), False),
+    ("AutoKey", (0, xrandr_raw_double), False),
+])
+def test_window_center_window(title, xrandr_raw_output, matchClass):
+    window = create_window()
+    xrandr = MagicMock(return_value = xrandr_raw_output)
+    with patch("autokey.scripting.window.Window.set_property") as set_prop:
+        with patch("autokey.scripting.window.Window.resize_move") as resize_move:
```

Comment:
[Don't need multiple `with`s](https://github.com/autokey/autokey/blob/29a8b86ecc0ed3e210b1845e2599b94a305aaf03/tests/engine_helpers.py#L43)

Reply:
Yeah I figured that there had to be some way to do it but was not aware of the syntax, neat!